shopping-list-web/src/AddItem/AddItem.tsx

```diff
@@ -0,0 +1,62 @@
+import { Dispatch, useState } from "react";
+import {
+  addCategoryDataForProduct,
+  getCategoryDataResponseForProduct,
+} from "../utilities/httpMethods/productMethods";
+import InputText from "./InputText/InputText";
+import InputCategory from "./InputCategory/InputCategory";
+
+export default function AddItem(props: {
+  currentProduct: string;
+  setCurrentProduct: Dispatch<React.SetStateAction<string>>;
+  currentAmount: string;
+  setCurrentAmount: Dispatch<React.SetStateAction<string>>;
+  handleItemSubmit: ((category: string) => Promise<void>) | (() => void);
+}) {
+  const [categoryFound, setCategoryFound] = useState(true);
+
+  const handleItemSubmit = async () => {
+    const category = await getCategoryDataForProduct(props.currentProduct);
+    if (category) {
+      await props.handleItemSubmit(category);
+    }
+  };
+
+  const handleCategorySubmit = async (category: string) => {
+    await addCategoryDataForProduct(props.currentProduct, category);
+    setCategoryFound(true);
+    await props.handleItemSubmit(category);
+  };
+
+  const getCategoryDataForProduct = async (product: string) => {
+    try {
+      const result = await getCategoryDataResponseForProduct(product);
+      if (result.error === "Product not found") {
+        setCategoryFound(false);
+        return;
+      }
+      return result.category;
+    } catch (error) {
+      console.error("Error: ", error);
+      return "miscellaneous"; //TODO: Improve this error handling
+    }
+  };
+
+  return (
+    <>
+      {categoryFound && (
+        <InputText
+          setCurrentProduct={props.setCurrentProduct}
+          setCurrentAmount={props.setCurrentAmount}
+          handleSubmit={handleItemSubmit}
```

Comment:
With my changes to input text that have gone in, this will need to change (2 more props for product and amount)